lib/core/src/Cardano/Wallet/DB/SqliteTypes.hs

```diff
@@ -0,0 +1,257 @@
+-- WalletId
+
+instance PersistField WalletId where
+    toPersistValue =
+        toPersistValue . toText
+    fromPersistValue pv = do
+        a <- fromText <$> fromPersistValue pv
+        case a of
+             Left _     ->
+                 Left $ "not a valid value: " <> T.pack (show pv)
+             Right txid ->
+                 pure txid
+
+instance PersistFieldSql WalletId where
+    sqlType _ = sqlType (Proxy @Text)
+
+instance Read WalletId where
+    readsPrec _ = error "readsPrec stub needed for persistent"
+
+instance ToHttpApiData WalletId where
+    toUrlPiece = toText
+
+instance FromHttpApiData WalletId where
+    parseUrlPiece = parseUrlPieceFromText
+
+instance ToJSON WalletId where
+    toJSON = String . toText
+
+instance FromJSON WalletId where
+    parseJSON = aesonFromText "WalletId"
+
+instance PathPiece WalletId where
+    fromPathPiece = fromTextMaybe
+    toPathPiece = toText
+
+
+----------------------------------------------------------------------------
+-- AddressScheme
+
+data AddressScheme = Sequential | Random | Any deriving (Show, Eq, Generic)
+
+instance PersistField AddressScheme where
+    toPersistValue = toPersistValue . toText
+    fromPersistValue pv = do
+        let err = "not a valid value: " <> T.pack (show pv)
+        first (const err) (fromPersistValue pv)
+
+instance PersistFieldSql AddressScheme where
+    sqlType _ = sqlType (Proxy @Text)
+
+instance FromText AddressScheme where
+    fromText txt = case txt of
+        "sequential" -> Right Sequential
+        "random" -> Right Random
+        "any" -> Right Any
+        _ ->
+            Left . TextDecodingError $ show txt
+                   <> " is neither \"sequential\", \"random\", nor \"any\""
+
+instance ToText AddressScheme where
+    toText Sequential = "sequential"
+    toText Random = "random"
+    toText Any = "any"
+
+----------------------------------------------------------------------------
+-- TxId
+
+-- Wraps Hash "Tx" because the persistent dsl doesn't like (Hash "Tx")
+newtype TxId = TxId { getTxId :: Hash "Tx" } deriving (Show, Eq, Ord, Generic)
+
+instance PersistField TxId where
+    toPersistValue = toPersistValue . toText . getTxId
+    fromPersistValue pv = do
+        a <- fromText <$> fromPersistValue pv
+        case a of
+             Left _     ->
+                 Left $ "not a valid value: " <> T.pack (show pv)
+             Right txid ->
+                 pure (TxId txid)
+
+instance PersistFieldSql TxId where
+    sqlType _ = sqlType (Proxy @Text)
+
+instance Read TxId where
+    readsPrec _ = error "readsPrec stub needed for persistent"
+
+instance ToJSON TxId where
+    toJSON = String . toText . getTxId
+
+instance FromJSON TxId where
+    parseJSON = fmap TxId . aesonFromText "WalletId"
+
+instance ToHttpApiData TxId where
+    toUrlPiece = toText . getTxId
+
+instance FromHttpApiData TxId where
+    parseUrlPiece = fmap TxId . parseUrlPieceFromText
```

Comment:
Same question about the JSON & HttpApiData instances?

Reply:
they are needed to derive `Database.Persist.Class.PersistEntity.Key TxMeta` and `Database.Persist.Class.PersistEntity.Key TxIn`